src/ui/wxWidgets/TreeCtrl.cpp

```diff
@@ -838,14 +839,50 @@ bool TreeCtrl::Remove(const CUUID &uuid)
 void TreeCtrl::SetItemImage(const wxTreeItemId &node,
                                const CItemData &item)
 {
-  // TODO: modify to display warning and expired states
+  int offset = 0;
+  CItemData::EntryType entrytype = item.GetEntryType();
+  
+  if((entrytype == CItemData::ET_NORMAL) || (entrytype == CItemData::ET_ALIASBASE) || (entrytype == CItemData::ET_SHORTCUTBASE)) {
+    time_t tttXTime(static_cast<time_t>(0));
+    item.GetXTime(tttXTime);
+    if (tttXTime > static_cast<time_t>(0) && tttXTime <= static_cast<time_t>(3650)) {
+      time_t tttCPMTime(static_cast<time_t>(0));
+      item.GetPMTime(tttCPMTime);
+      if (tttCPMTime == static_cast<time_t>(0))
+        item.GetCTime(tttCPMTime);
+      tttXTime = static_cast<time_t>((long)tttCPMTime + (long)tttXTime * 86400L);
```

Comment:
What's happening on this line? Is it trying to correct an invalid tttXTime?

Reply:
Add comment about handling of time versus handling of distance of days before expiry.